src/Compilers/Core/Portable/SourceGeneration/Nodes/DriverStateTable.cs

```diff
@@ -84,6 +98,13 @@ public IStateTable GetSyntaxInputTable(ISyntaxInputNode syntaxInputNode)
                     builders.Free();
                 }
 
+                // if we don't have an entry for this node, it must have thrown an exception
+                if (!_tableBuilder.ContainsKey(syntaxInputNode))
+                {
+                    Debug.Assert(_syntaxExceptions.ContainsKey(syntaxInputNode));
```

Comment:
>Debug.Assert(_syntaxExceptions.ContainsKey(syntaxInputNode)); [](http://example.com/codeflow?start=20&length=61)

The assert seems unnecessary since the dictionary access below will throw an exception if the assert would have failed.

Reply:
It was just a statement of intent to show the thing below *wont* throw, but happy to remove.